content/blog/2022-05-18-learned-federated-query-optimization.md

```diff
@@ -0,0 +1,178 @@
++++
+title = "A Learned Query Compiler"
+[extra]
+latex = true
+[[extra.authors]]
+name = "Victor Giannakouris"
+link = "https://www.linkedin.com/in/vgian/"
++++
+
+# What was the goal?
+The goal of this project was to create an optimizer for a cross-database query compiler. A cross-database query, also 
+known as federated query, is an SQL query that consists of tables that span across different, individual database 
+systems. The main weakness with the already existing federated query optimization approaches is of the complexity
+of integration, as the existing solutions need to obtain specific knowledge from the externally connected databases
+by writing complex drivers. We propose a solution based on a machine learning model that enables the easy integration of
```

Comment:
I think this sentence could be even clearer about what the problem is. Namely, this kind of design fundamentally needs to know things about the underlying engines: table sizes, relative storage speeds, etc. Existing systems get those manually. We want to get them automatically. (But giving a clear idea of _what_ that integration must consist of would help a lot.)